l10n_ch_import_winbiz/wizard/l10n_ch_import_winbiz.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author Vincent Renaville
+#    Copyright 2015 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import sys
+import traceback
+import logging
+from lxml import etree
+from StringIO import StringIO
+from openerp import models, fields, api
+from openerp.tools.translate import _
+from itertools import izip_longest
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountWinbizImport(models.TransientModel):
+    _name = 'account.winbiz.import'
+    _description = 'Import Accounting Winbiz'
+
+    company_id = fields.Many2one('res.company', 'Company',
+                                 invisible=True)
+    period_id = fields.Many2one('account.period', 'Period',
+                                required=True)
+    report = fields.Text(
+        'Report',
+        readonly=True
+        )
+    journal_id = fields.Many2one('account.journal', 'Journal',
+                                 required=True)
+    state = fields.Char(sting="Import state"
+                        'Report',
+                        readonly=True,
+                        default="draft"
+                        )
+    file = fields.Binary(
+        'File',
+        required=True
+        )
+    imported_move_ids = fields.Many2many(
+        'account.move', 'import_cresus_move_rel',
+        string='Imported moves')
+
+    help_html = fields.Html('Import help', readonly=True,
+                            default=_('''
+                 In order to import your 'Winbiz Salaires' .xml \
+                 file you must complete the following requirements : </br>
+                * The accounts, analytical accounts used in the Cresus\
+                 file must be previously created into Odoo  </br>
+                * The date of the entry will determine the period used\
+                 in Odoo, so please ensure the period is created already. </br>
+                </br>'''))
```

Comment:
On 29/07/2015 11:20, Vincent Renaville wrote:
> In l10n_ch_import_winbiz/wizard/l10n_ch_import_winbiz.py
> <https://github.com/OCA/l10n-switzerland/pull/156#discussion_r35741238>:
> 
>> +        'File',
>> +        required=True
>> +        )
>> +    imported_move_ids = fields.Many2many(
>> +        'account.move', 'import_cresus_move_rel',
>> +        string='Imported moves')
>> +
>> +    help_html = fields.Html('Import help', readonly=True,
>> +                            default=_('''
>> +                 In order to import your 'Winbiz Salaires' .xml \
>> +                 file you must complete the following requirements : </br>
>> +                * The accounts, analytical accounts used in the Cresus\
>> +                 file must be previously created into Odoo  </br>
>> +                * The date of the entry will determine the period used\
>> +                 in Odoo, so please ensure the period is created already. </br>
>> +                </br>'''))
> 
> @gurneyalex <https://github.com/gurneyalex> I will change the view * and
> br but your can find the field there in the form view:
> https://github.com/OCA/l10n-switzerland/pull/156/files#diff-d2f02452a39d948df29350d804fcce9bR41

My question is: why use a field here rather than a Label?


-- 
Alexandre Fayolle
Chef de Projet
Tel : +33 4 58 48 20 30

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac Cedex
http://www.camptocamp.com